src/view/InputBox.tsx

```diff
@@ -1,14 +1,80 @@
 import React from 'react';
+import ReactCodeMirror from 'react-codemirror';
+import CodeMirror from 'codemirror';
+require('codemirror/lib/codemirror.css');
+require('./InputBox.css');
+require('codemirror/mode/python/python');
 
-const inputBoxStyle: React.CSSProperties = {
-  borderRadius: 24,
-  paddingLeft: 8,
-  paddingRight: 8,
-  fontSize: 24,
-  textAlign: 'center',
-  verticalAlign: 'middle',
-  outline: 'none',
-};
+CodeMirror.defineMode('boolean-expressions', function(config, modeOptions) {
+  const KEYWORDS = new Set([
+    'and',
+    'not',
+    'or',
+    'xor',
+    'if',
+    'then',
+    'iff',
+    'only',
+    'true',
+    'false',
+    'equals',
+    'imply',
+    'implies',
+    'nimply',
+    'nimplies',
+    'xnor',
+    'nand',
+    'nor',
+    '<-',
+    '<->',
+    '->',
+    '<=',
+    '=>',
+    '<=>',
+    '<--',
+    '-->',
+    '<-->',
+    '<==>',
+    '<==',
+    '==>',
+    '=',
+    '==',
+    '<=/=',
+    '=/=>',
+    '<-/-',
+    '-/->',
+    '&',
+    '&&',
+    '|',
+    '||',
+    '=/=',
+    '!=',
+    '~=',
+    '^',
+    '\\/',
+    '/\\',
+    '!',
+    '~',
+  ]);
+  return {
+    token: function(stream, state) {
+      const ch = stream.next();
+      if (ch === ' ') {
+        return null;
+      }
+      while (stream.peek() && stream.peek() !== ' ') {
+        stream.next();
+      }
+      const word: string = stream.current();
+      if (KEYWORDS.has(word)) {
+        stream.eat('word: ' + word);
+        return 'keyword';
+      }
+      stream.eat(word);
+      return null;
+    },
+  };
+});
```

Comment:
yeah yeah you probably want this bad boy in it's own file, huh? Maybe deduplicate this keywords set throughout the code base?

Reply:
yeah prolly 😝